packages/lesswrong/lib/collections/comments/schema.ts

```diff
@@ -875,7 +875,58 @@ const schema: SchemaType<"Comments"> = {
     canRead: ['guests'],
     canUpdate: ['sunshineRegiment', 'admins'],
     canCreate: ['members', 'sunshineRegiment', 'admins'],
-  } 
+  },
+
+  doppelComments: {
+    // Resolved by doppelComments resolver
+    type: Array,
+    optional: true,
+    canRead: ['guests'],
+    canCreate: ['admins'],
+    canUpdate: ['admins'],
+    hidden: true,
+  },
+
+  'doppelComments.$': {
+    type: Object,
+  },
+
+  ownDoppelCommentVote: resolverOnlyField({
+    type: Object,
+    graphQLtype: 'DoppelCommentVote',
+    canRead: ['guests'],
+    resolver: async (comment: DbComment, args: void, context: ResolverContext) => {
+      const { DoppelCommentVotes, currentUser } = context;
+      if (!currentUser) return null;
+      return DoppelCommentVotes.findOne({commentId: comment._id, userId: currentUser._id}) ?? null;
+    },
+    sqlResolver: ({field, currentUserField}) => `(
+      SELECT ROW_TO_JSON(dc.*)
+      FROM "DoppelCommentVotes" dc
+      WHERE dc."commentId" = ${field("_id")} AND dc."userId" = ${currentUserField("_id")}
+    )`
+  }),
+  doppelCommentVoteChoices: resolverOnlyField({
+    type: Array,
+    graphQLtype: '[String]',
+    canRead: ['guests'],
+    resolver: async (comment: DbComment, args: void, context: ResolverContext) => {
+      const { DoppelCommentVotes } = context;
+      const doppelCommentVotes = await DoppelCommentVotes.find({commentId: comment._id, type: 'vote'}).fetch();
+      return doppelCommentVotes.map(vote => vote.doppelCommentChoiceId);
+    },
+    sqlResolver: ({field}) => `(
+      SELECT ARRAY_AGG(dc."doppelCommentChoiceId")
+      FROM "DoppelCommentVotes" dc
+      WHERE dc."commentId" = ${field("_id")} AND dc."type" = 'vote'
+      GROUP BY dc."commentId"
+    )`
+  }),
+  "doppelCommentVoteChoices.$": {
+    type: String,
+    optional: true,
+    nullable: true,
```

Comment:
...should this in fact be `nullable`?  `doppelCommentChoiceId` looks like it is nullable; are we intending to return `null` values inside of `doppelCommentVoteChoices` that are unattached to  anything in particular?  (I forget, maybe the cases where that's null are in cases where the type is not `vote`?  In which case we don't want to return `null` values, and if we do we're violating an implicit invariant; probably we want to filter those out anyways.)